### ISSUE TITLE
Beach should be rendered under water

### ISSUE THREAD

**Issue by biodranik** [organicmaps/organicmaps#4245]
https://www.openstreetmap.org/#map=17/58.49051/-6.22575

| OM | OSM | OSM |
|----|-----|-----|
|![image](https://user-images.githubusercontent.com/170263/211272371-d1e64d29-f113-4ac6-ab16-7c1a20a466bc.png)|![image](https://user-images.githubusercontent.com/170263/211272392-0c93c1b1-0dc8-4ef5-be5b-1ba2b395f46e.png)|![image](https://user-images.githubusercontent.com/170263/211272417-c1f5315d-51b8-4fc2-9020-b9653582cb73.png)|

**Comment by Jean-BaptisteC** [comment]
CC @pastk

**Comment by pastk** [comment]
A big part of the beach is beyond the coastline.
ATM ocean/sea water has the lowest priority and hence all landcover is rendered above it, like this beach.

Ideally the tidal part of the beach beyond the coastline should be rendered "underwater" with some transparency.

**Comment by pastk** [comment]
We can experiment with moving coastline water to the same "bg-top" priority range as other areal water, i.e. higher than landcover areas. But likely it'll lead to e.g. some islands disappearing if they were mapped just over water w/o using multipolygon "holes".

**Comment by pastk** [comment]
Its a part of a bigger picture #5377

**Comment by pastk** [comment]
> But likely it'll lead to e.g. some islands disappearing if they were mapped just over water w/o using multipolygon "holes"

E.g. this off-shore beach island https://www.openstreetmap.org/way/4516234 is not visible on osm.org but visible in OM.

**Comment by biodranik** [comment]
> Ideally the tidal part of the beach beyond the coastline should be rendered "underwater" with some transparency.

That would be great! What is needed to implement it?

**Comment by pastk** [comment]
> 
> That would be great! What is needed to implement it?

We need to fix #5339 first for sure, otherwise transparency won't work.

**Comment by biodranik** [comment]
Good point. Is there a validator for these mapping issues? Is there an easy to use online tool for inexperienced mappers and OM users to quickly check for mapping errors?

**Comment by map-per** [comment]
I don't think so, at least I don't know any tool.

**Comment by pastk** [comment]
> Would it be better to draw the ways as well over river areas? or avoid due to style?

It was like that for some time due to an unnoticed side-effect of some change, then fixed in #4927 :)

Generally it doesn't make sense and sometimes looks weird (in big rivers especially).

**Comment by patepelo** [comment]
> Generally it doesn't make sense and sometimes looks weird (in big rivers especially).

Due to style, I'm more in favor of the current representation but it may be useful for a[ future river routing](https://github.com/organicmaps/organicmaps/issues/823).

Although render is not needed for routing, due to data not being consistent, it may be of help to understand proposed routing.

**Comment by patepelo** [comment]
Osmand has those and waterway routing as well


![Screenshot_20231123-140330.png](https://github.com/organicmaps/organicmaps/assets/23003127/3254a2fe-c306-4a00-b185-6ef6c29c1dcd)
